servicenow/icon_servicenow/triggers/incident_created/trigger.py

```diff
@@ -0,0 +1,76 @@
+import insightconnect_plugin_runtime
+import time
+from .schema import IncidentCreatedInput, IncidentCreatedOutput, Input, Output, Component
+# Custom imports below
+from insightconnect_plugin_runtime.exceptions import PluginException
+from datetime import datetime, timezone, timedelta
+import pytz
+
+
+class IncidentCreated(insightconnect_plugin_runtime.Trigger):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+                name='incident_created',
+                description=Component.DESCRIPTION,
+                input=IncidentCreatedInput(),
+                output=IncidentCreatedOutput())
+        self.last_poll_time = (datetime.now(tz=timezone.utc) - timedelta(seconds=1)).replace(microsecond=0)
+        self.most_recent_system_id = ""
+        self.last_sys_id = ""
+
+    def poll(self, url, method, query, utc):
+        response = self.connection.request.make_request(url, method, params=query)
+
+        try:
+            results = response["resource"].get("result")
+        except KeyError as e:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN,
+                                  data=response.text) from e
+
+        incidents = [(result.get("sys_created_on"), result.get("sys_id")) for result in results
+                     if result.get("sys_created_on") is not None and result.get("sys_id") is not None]
+
+        # Incidents stored from least to most recent date
+        # Loop from most recent incident to least recent incident
+        for i in range(len(incidents)-1, -1, -1):
+            incident = incidents[i]
+            d = datetime.strptime(incident[0], "%Y-%m-%d %H:%M:%S")
```

Comment:
Are the indices in incident guaranteed (same as on line 43)? Error handling around these may be worthwhile in the case of an unexpected response from SNOW. One thing I've learned is that APIs can't be trusted ;)